spec/unit/logger_spec.rb

```diff
@@ -42,5 +42,21 @@ module Watir
       message = /WARN Watir \[DEPRECATION\] #old is deprecated\. Use #new instead\./
       expect { Watir.logger.deprecate('#old', '#new') }.to output(message).to_stdout
     end
+
+    it 'allows to selectively ignore warnings with Strings' do
+      Watir.logger.ignore("old deprecated")
+      expect { Watir.logger.deprecate('#old', '#new', ids: ['old deprecated']) }.to_not output.to_stdout
+    end
```

Comment:
I'm not sure I understand the intent here. There don't appear to be any String ids at the moment. Is this for a future usage? Wouldn't it be weird to mix String and Symbol ids?

Reply:
The idea is to allow other implementations to use String or Symbol instances. Not certain it's the best idea, the updated commits make my intention more clear for my current desired behavior.